src/parquet/column/reader.cc

```diff
@@ -33,7 +33,10 @@ ColumnReader::ColumnReader(const ColumnDescriptor* descr,
   : descr_(descr),
     pager_(std::move(pager)),
     num_buffered_values_(0),
-    num_decoded_values_(0) {}
+    num_decoded_values_(0) {
+      repetition_level_decoder_.Init(descr->max_repetition_level());
+      definition_level_decoder_.Init(descr->max_definition_level());
```

Comment:
Thoughts on combining `Init` and `Reset` (so this could be removed here), and `Reset` would have to add a null check on the decoder of interest? Then these could be removed